tables_io/io_layer.py

```diff
@@ -9,24 +10,28 @@
     from astropy.io import fits
     HAS_ASTROPY = True
 except ImportError:  #pragma: no cover
+    print("Warning: astropy.table is not available.  tables_io will still work with other formats.")
     HAS_ASTROPY = False
 
 try:
     import h5py
     HAS_HDF5 = True
 except ImportError:  #pragma: no cover
+    print("Warning: h5py is not available.  tables_io will still work with other formats.")
     HAS_HDF5 = False
 
 try:
     import pandas as pd
     HAS_PANDAS = True
 except ImportError:  #pragma: no cover
+    print("Warning: pandas is not available.  tables_io will still work with other formats.")
     HAS_PANDAS = False
 
 try:
     import pyarrow.parquet as pq
```

Comment:
One other thing is that as the list of libraries used gets larger you tend to spend quite a lot of time importing things that you may not actually use.  One option to avoid this is to do the imports in the methods that use them - that works okay if they are relatively localised but is annoying otherwise.

There's also some lazy importing methods, e.g. this one:
https://stackoverflow.com/questions/42703908/how-do-i-use-importlib-lazyloader

Reply:
The lazy import seems ok.   Might make sense to collect all the importing cleverness into one place.  I agree that local imports are annoying.